extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -839,6 +839,34 @@ export class ProjectsController {
 		}
 	}
 
+	public async rename(context: dataworkspace.WorkspaceTreeItem): Promise<void> {
+		const node = context.element as BaseProjectTreeItem;
+		const project = this.getProjectFromContext(node);
+		const file = this.getFileProjectEntry(project, node);
+
+		// need to use quickpick because input box isn't supported in treeviews
+		// https://github.com/microsoft/vscode/issues/117502 and https://github.com/microsoft/vscode/issues/97190
+		const newFileName = await vscode.window.showInputBox(
+			{
+				title: constants.enterNewName,
+				value: path.basename(node.friendlyName, constants.sqlFileExtension),
+				ignoreFocusOut: true
+			});
+
+		if (!newFileName) {
+			return;
+		}
+
+		// TODO: swap this out and hookup to "Move" file/folder api
+		// rename the file
+		const newFilePath = path.join(path.dirname(file?.fsUri.fsPath!), `${newFileName}.sql`);
+		await fs.rename(file?.fsUri.fsPath!, newFilePath);
```

Comment:
What will happen if a file with same name exists?

Reply:
Added validation: 
![image](https://user-images.githubusercontent.com/31145923/217096470-0d305931-8b5c-4248-a8c6-5d11e82d95b5.png)

Reply:
I don't see any code changes, any chance you might have forgotten to push the changes.

Reply:
whoops pushed the changes now